controller/gke-cluster-config-handler.go

```diff
@@ -127,24 +173,27 @@ func (h *Handler) OnGkeConfigRemoved(key string, config *gkev1.GKEClusterConfig)
 		return config, nil
 	}
 
-	logrus.Infof("deleting cluster [%s]", config.Name)
-
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	svc, err := utils.GetServiceClient(ctx, config.Spec.CredentialContent)
+	cred, err := getSecret(ctx, h.secretsCache, &config.Spec)
+	if err != nil {
+		return config, err
+	}
+	client, err := gke.GetGKEClient(ctx, cred)
 	if err != nil {
 		return config, err
 	}
 
-	logrus.Debugf("Removing cluster %v from project %v, region/zone %v", config.Spec.ClusterName, config.Spec.ProjectID, config.Spec.Region)
-	operation, err := utils.WaitClusterRemoveExp(ctx, svc, config)
-	if err != nil && !strings.Contains(err.Error(), "notFound") {
+	logrus.Infof("removing cluster %v from project %v, region/zone %v", config.Spec.ClusterName, config.Spec.ProjectID, config.Spec.Region)
+	if err := gke.RemoveCluster(ctx, client, config); err != nil {
+		logrus.Debugf("error deleting cluster %s: %v", config.Spec.ClusterName, err)
+		return config, err
+	}
+
+	if err := h.deleteCASecret(config); err != nil {
```

Comment:
@rmweir I don't think eks-operator does this, is there a reason why?

Reply:
I think it adds the config as an owner reference of the secret so it gets deleted automatically.

Reply:
Looks like you did the same, should be able to remove this. If you delete config you should still observe the secret being deleted.

Reply:
Thanks, removed